includes/Abilities/Summarization/Summarization.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * Content summarization WordPress Ability implementation.
+ *
+ * @package WordPress\AI
+ */
+
+declare( strict_types=1 );
+
+namespace WordPress\AI\Abilities\Summarization;
+
+use WP_Error;
+use WordPress\AI\Abstracts\Abstract_Ability;
+use WordPress\AI_Client\AI_Client;
+
+use function WordPress\AI\get_post_context;
+use function WordPress\AI\get_preferred_models;
+use function WordPress\AI\normalize_content;
+
+/**
+ * Content summarization WordPress Ability.
+ *
+ * @since x.x.x
+ */
+class Summarization extends Abstract_Ability {
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @since x.x.x
+	 */
+	protected function input_schema(): array {
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'content' => array(
+					'type'              => 'string',
+					'sanitize_callback' => 'sanitize_text_field',
+					'description'       => esc_html__( 'Content to summarize.', 'ai' ),
+				),
+				'post_id' => array(
+					'type'              => 'integer',
+					'sanitize_callback' => 'absint',
+					'description'       => esc_html__( 'Content from this post will be summarized. This overrides the content parameter if both are provided.', 'ai' ),
+				),
```

Comment:
I've been thinking about ways to make abilities more generalized for broader use. What would you think about renaming this to `context` and allowing _either_ a string or integer? If it's an integer, it assumes it's a post ID and does what it's doing now. If it's a context, then it's simply additional details provided to the AI, making it clear in the prompt that it should not be part of the summary.

Reply:
Thinking about this more I think I'm following what you're suggesting now.

Right now if someone wanted to manually pass in content that isn't associated with any post and they also wanted additional context with that, they could do that all in the content field but that's probably not obvious to anyone wanting to do that.

Whereas if we change `post_id` to `context`, we can make it clear that you can pass in a post ID and it will automatically grab context from that or you can pass in your own additional context that will end up in the request.

I can go ahead and make that change